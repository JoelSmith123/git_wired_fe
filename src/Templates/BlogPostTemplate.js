import React, { Component } from 'react';
import './Templates.css';

export default class BlogPostTemplate extends Component {
  constructor() {
    super()

    this.state = {
      headerButtons: 0
    }
  }

  addBtnToHeader = (event) => {
    event.preventDefault()
    this.setState({ headerButtons: this.state.headerButtons + 1 })
  }

  renderBtnToHeader = () => {
    const btnArr = []
    for (let i = 0; this.state.headerButtons > i; i++) {
      btnArr.push(<button key={i} className='template-header-btn'></button>)
    }
    return btnArr
  }

  render() {
    return (
      <div className='BlogPostTemplate'>
        <div className='template-header'>
        <div className='pseudo-btn-space'></div>
          <i className="far fa-plus-square" onClick={this.addBtnToHeader}></i>
          {
            this.renderBtnToHeader()
          }
        </div>
        <div className='template-card-container'>
          
        </div>
        <div className='template-footer'></div>
      </div>
    )
  }
}
