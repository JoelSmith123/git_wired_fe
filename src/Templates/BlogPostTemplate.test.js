import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import BlogPostTemplate from './BlogPostTemplate.js'

describe('BlogPostTemplate', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BlogPostTemplate />, div);
    ReactDOM.unmountComponentAtNode(div);
  });  

  it('should match snapshot', () => {
    const wrapper = shallow(<BlogPostTemplate />)
    expect(wrapper).toMatchSnapshot()
  })   

  it('should update state with new header buttons', () => {
    const wrapper = shallow(<BlogPostTemplate />)
    const mockEvent = { target: {}, preventDefault: jest.fn() }

    wrapper.setState({ headerButtons: 1 })
    wrapper.instance().addBtnToHeader(mockEvent)

    expect(wrapper.state('headerButtons')).toEqual(2)
  })

  it('should render new buttons based on state', () => {
    const wrapper = shallow(<BlogPostTemplate />)
    const mockHeaderButtonArr = "[{\"type\":\"button\",\"key\":\"0\",\"ref\":null,\"props\":{\"className\":\"template-header-btn\"},\"_owner\":null,\"_store\":{}}]"
    wrapper.setState({ headerButtons: 1 })

    expect(JSON.stringify(wrapper.instance().renderBtnToHeader())).toEqual(mockHeaderButtonArr)
  })
})