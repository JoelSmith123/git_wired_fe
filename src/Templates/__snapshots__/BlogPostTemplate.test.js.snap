// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostTemplate should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlogPostTemplate />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="template-header"
        >
          <div
            className="pseudo-btn-space"
          />
          <i
            className="far fa-plus-square"
            onClick={[Function]}
          />
        </div>,
        <div
          className="template-card-container"
        >
          <div
            className="template-post-main"
          />
          <div
            className="template-post-sub-container"
          >
            <div
              className="template-post-sub"
            />
            <div
              className="template-post-sub"
            />
          </div>
        </div>,
        <div
          className="template-footer"
        />,
      ],
      "className": "BlogPostTemplate",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="pseudo-btn-space"
            />,
            <i
              className="far fa-plus-square"
              onClick={[Function]}
            />,
            Array [],
          ],
          "className": "template-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pseudo-btn-space",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "far fa-plus-square",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="template-post-main"
            />,
            <div
              className="template-post-sub-container"
            >
              <div
                className="template-post-sub"
              />
              <div
                className="template-post-sub"
              />
            </div>,
          ],
          "className": "template-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "template-post-main",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="template-post-sub"
                />,
                <div
                  className="template-post-sub"
                />,
              ],
              "className": "template-post-sub-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "template-post-sub",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "template-post-sub",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "template-footer",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="template-header"
          >
            <div
              className="pseudo-btn-space"
            />
            <i
              className="far fa-plus-square"
              onClick={[Function]}
            />
          </div>,
          <div
            className="template-card-container"
          >
            <div
              className="template-post-main"
            />
            <div
              className="template-post-sub-container"
            >
              <div
                className="template-post-sub"
              />
              <div
                className="template-post-sub"
              />
            </div>
          </div>,
          <div
            className="template-footer"
          />,
        ],
        "className": "BlogPostTemplate",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="pseudo-btn-space"
              />,
              <i
                className="far fa-plus-square"
                onClick={[Function]}
              />,
              Array [],
            ],
            "className": "template-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pseudo-btn-space",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "far fa-plus-square",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="template-post-main"
              />,
              <div
                className="template-post-sub-container"
              >
                <div
                  className="template-post-sub"
                />
                <div
                  className="template-post-sub"
                />
              </div>,
            ],
            "className": "template-card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "template-post-main",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="template-post-sub"
                  />,
                  <div
                    className="template-post-sub"
                  />,
                ],
                "className": "template-post-sub-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "template-post-sub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "template-post-sub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "template-footer",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
